central/src/main/webapp/components/ModelNavigator.jsx

```diff
@@ -86,14 +98,99 @@ export default function ModelNavigator(props) {
+                if ((versionValue == '') || (versionValue == model.version)){
+                    if ((typeValue == '') || (typeValue == application.key)){
+                        return model.name.toLowerCase().includes(nameValue.toLowerCase());
+                    }
+                }
+            })
+        ))
+
+	const modelNameFilterOnChange = (event) => {
+        setNameValue(event.target.value);
+        setModelList(nameFilteredModels)
+
+    };
+
+    const modelTypeFilterOnChange = (event) => {
+        setTypeValue(event.target.value);
+    };
+
+    const modelVersionFilterOnChange = (event) => {
+        setVersionValue(event.target.value);
+    };
 
 	return (
 		<>
+           <div className={classes.view_root}>
+                <div className={classes.navigator_root}>
+                    <TreeView
+                        defaultCollapseIcon={<ExpandMoreIcon />}
+                        defaultExpandIcon={<ChevronRightIcon />}
+                    >
+                        <TreeItem nodeId="Model Searching" label="Model Search">
+                            <TreeView
+                                defaultCollapseIcon={<ExpandMoreIcon />}
+                                defaultExpandIcon={<ChevronRightIcon />}
+                            >
+                                <TreeItem nodeId="Name" label="Name">
+                                    <TextField
+                                        id="name-search"
+                                        label="Enter Name"
+                                        value={nameValue}
+                                        onChange={modelNameFilterOnChange}
+                                        inputProps={{style: { backgroundColor:"white"},}}
+                                    />
+                                </TreeItem>
+                                <TreeItem nodeId="Version" label="Version">
+                                    <div onChange={modelVersionFilterOnChange}>
+                                        <button disabled={versionValue=="0.0.1"} value="0.0.1" onClick={modelVersionFilterOnChange} > 0.0.1 </button>
+                                        <button disabled={versionValue=="0.0.2"} value="0.0.2" onClick={modelVersionFilterOnChange} > 0.0.2 </button>
+                                        <button disabled={versionValue=="0.0.3"} value="0.0.3" onClick={modelVersionFilterOnChange} > 0.0.3 </button>
+                                        <button value="" onClick={modelVersionFilterOnChange} > Clear </button>
+                                    </div>
+                                </TreeItem>
+                                <TreeItem nodeId="Type" label="Type">
+                                    <div onChange={modelVersionFilterOnChange}>
+                                       <button disabled={typeValue=='CV.ACTION_RECOGNITION'} value= 'CV.ACTION_RECOGNITION' onClick={modelTypeFilterOnChange} > action_recognition </button>
```

Comment:
Can you load the list of applications (and versions) from modelZooData rather than hardcoding it?